File: backend/app/models/ai.py
"""Pydantic schemas for generative AI endpoints."""
from __future__ import annotations

from pydantic import BaseModel, Field


class ChatStreamRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=8000)


class AIFeedbackRequest(BaseModel):
    """Human correction for AI-suggested labels (e.g. category) stored in pgvector."""

    content_text: str = Field(..., min_length=1, max_length=16_000)
    suggested_label: str = Field(..., min_length=1, max_length=256)
    corrected_label: str = Field(..., min_length=1, max_length=256)
    source: str = Field(default="ui", max_length=64)


class AIFeedbackResponse(BaseModel):
    embedding_id: str
    message: str = "Feedback stored."

File: frontend/lib/schemas/ai.ts
import { z } from "zod";

export const aiFeedbackRequestSchema = z.object({
  content_text: z.string().min(1).max(16000),
  suggested_label: z.string().min(1).max(256),
  corrected_label: z.string().min(1).max(256),
  source: z.string().max(64).optional().default("ui"),
});

export const aiFeedbackResponseSchema = z.object({
  embedding_id: z.string().uuid(),
  message: z.string().optional(),
});

export type AiFeedbackRequest = z.infer<typeof aiFeedbackRequestSchema>;
export type AiFeedbackResponse = z.infer<typeof aiFeedbackResponseSchema>;
